packages/cosmic-proto/package.json

```diff
@@ -67,6 +71,10 @@
     "./cosmos/staking/v1beta1/tx.js": {
       "types": "./dist/codegen/cosmos/staking/v1beta1/tx.d.ts",
       "default": "./dist/codegen/cosmos/staking/v1beta1/tx.js"
+    },
+    "./cosmos/distribution/v1beta1/tx.js": {
+      "types": "./dist/codegen/cosmos/distribution/v1beta1/tx.d.ts",
+      "default": "./dist/codegen/cosmos/distribution/v1beta1/tx.js"
```

Comment:
Probably just need one or the other - this is unfortunately not being done consistently in the existing exports:
- "./cosmos/distribution/v1beta1/tx.js"
- "./cosmos/distribution/v1beta1/tx"

Also would be nice if we didn't need to specify these manually - my current attempts at globbed exports were unsuccessful.

Reply:
We could export just `cosmos` and use destructuring to get the specific module. Of course, that has the potential for bloat because Endo doesn't tree shake. We could start with the easy way and before release see whether it's worth the optimization of specifying each module.

Reply:
> my current attempts

Not my area of expertise but this was the attempt:

```diff
diff --git a/packages/cosmic-proto/package.json b/packages/cosmic-proto/package.json
index 36235b417..0d09641f5 100644
--- a/packages/cosmic-proto/package.json
+++ b/packages/cosmic-proto/package.json
@@ -28,45 +28,13 @@
       "types": "./dist/codegen/cosmos/bundle.d.ts",
       "default": "./dist/codegen/cosmos/bundle.js"
     },
-    "./cosmos/staking/v1beta1/tx": {
-      "types": "./dist/codegen/cosmos/staking/v1beta1/tx.d.ts",
-      "default": "./dist/codegen/cosmos/staking/v1beta1/tx.js"
+    "./*/*": {
+      "types": "./dist/codegen/*/*/*.d.ts",
+      "default": "./dist/codegen/*/*/*.js"
     },
-    "./ibc/applications/transfer/v1/tx.js": {
-      "types": "./dist/codegen/ibc/applications/transfer/v1/tx.d.ts",
-      "default": "./dist/codegen/ibc/applications/transfer/v1/tx.js"
-    },
-    "./google/protobuf/any": {
-      "types": "./dist/codegen/google/protobuf/any.d.ts",
-      "default": "./dist/codegen/google/protobuf/any.js"
-    },
-    "./swingset/msgs.js": {
-      "types": "./dist/codegen/agoric/swingset/msgs.d.ts",
-      "default": "./dist/codegen/agoric/swingset/msgs.js"
-    },
-    "./swingset/query.js": {
-      "types": "./dist/codegen/agoric/swingset/query.d.ts",
-      "default": "./dist/codegen/agoric/swingset/query.js"
-    },
-    "./swingset/swingset.js": {
-      "types": "./dist/codegen/agoric/swingset/swingset.d.ts",
-      "default": "./dist/codegen/agoric/swingset/swingset.js"
-    },
-    "./vstorage/query.js": {
-      "types": "./dist/codegen/agoric/vstorage/query.d.ts",
-      "default": "./dist/codegen/agoric/vstorage/query.js"
-    },
-    "./cosmos/tx/v1beta1/tx.js": {
-      "types": "./dist/codegen/cosmos/tx/v1beta1/tx.d.ts",
-      "default": "./dist/codegen/cosmos/tx/v1beta1/tx.js"
-    },
-    "./cosmos/staking/v1beta1/staking.js": {
-      "types": "./dist/codegen/cosmos/staking/v1beta1/staking.d.ts",
-      "default": "./dist/codegen/cosmos/staking/v1beta1/staking.js"
-    },
-    "./cosmos/staking/v1beta1/tx.js": {
-      "types": "./dist/codegen/cosmos/staking/v1beta1/tx.d.ts",
-      "default": "./dist/codegen/cosmos/staking/v1beta1/tx.js"
+    "./*/*/*": {
+      "types": "./dist/codegen/*/*/*/*.d.ts",
+      "default": "./dist/codegen/*/*/*/*.js"
     }
   },
```

RE bloat - so far seeing reasonable compressed bundle sizes for `init-orchestration.js` (0.34MB + 0.20MB) and `init-stakeAtom.js` (0.42MB + 0.20MB)

Reply:
Unfortunately the exports field only supports one wildcard per path. This works,
```json
    "./cosmos/tx/v1beta1/*.js": {
      "types": "./dist/codegen/cosmos/tx/v1beta1/*.d.ts",
      "default": "./dist/codegen/cosmos/tx/v1beta1/*.js"
    },
    "./cosmos/staking/v1beta1/*.js": {
      "types": "./dist/codegen/cosmos/staking/v1beta1/*.d.ts",
      "default": "./dist/codegen/cosmos/staking/v1beta1/*.js"
    }
```

but doesn't save us much boilerplate.

Reply:
> Probably just need one or the other

OK. going with just `./cosmos/distribution/v1beta1/tx.js`

Reply:
We were just holding it wrong. https://github.com/Agoric/agoric-sdk/pull/9310 has it working. I'll resolve this conversation and we can tackle this wildcard question in that issue